src/main/java/seedu/address/ui/AssignmentCardWithSchedule.java

```diff
@@ -0,0 +1,144 @@
+package seedu.address.ui;
+
+import static seedu.address.model.assignment.Deadline.DEADLINE_DATE_TIME_FORMAT;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.ResolverStyle;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.assignment.Assignment;
+import seedu.address.model.assignment.Deadline;
+
+/**
+ * An UI component that displays information of a {@code Assignment}.
+ */
+public class AssignmentCardWithSchedule extends UiPart<Region> {
+    private static final long MIN_PER_HOUR = 60;
+    private static final long HOUR_PER_DAY = 24;
+    private static final long DAY_PER_WEEK = 7;
+    private static final String FXML = "AssignmentListCardWithSchedule.fxml";
+    private static final String OVERDUE_STYLE_CLASS = "overdue";
+    private static final String DUE_SOON_STYLE_CLASS = "due-soon";
+    private static final String DUE_IN_A_WEEK_STYLE_CLASS = "due-in-a-week";
+    private static final DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern(DEADLINE_DATE_TIME_FORMAT)
+            .withResolverStyle(ResolverStyle.STRICT);
```

Comment:
what's this ResolverStyle just curious haha

Reply:
It just for the uuuu in YearOfEra